### DISCUSSION TITLE
Forked rebuild & upgraded to FastAPI 0.88, SQLAlchemy 1.4, & Nuxt 3.0

### DISCUSSION THREAD

**Discussion by turukawa** [fastapi/full-stack-fastapi-template#1806] [Questions]
I [forked this repo](https://github.com/whythawk/full-stack-fastapi-postgresql) more than a year ago for a major project and, with help from @br3ndonland, consolidated it on his [Inboard FastAPI images](https://inboard.bws.bio/docker), added in support for Neo4j (so I could use it for search), and fixed some of the things what were broken. 

With several big generational updates on core dependencies, I've spent the last two weeks updating and refactoring. All dependencies upgraded to the latest long-term support & working versions, and I built a tiny blog-type app as the base stack. Still a Cookiecutter template.

**Key updates:**

- Inboard 0.10.4 -> 0.37.0, including FastAPI 0.88
- SQLAlchemy 1.3 -> 1.4
- Celery 4.4 -> 5.2
- Postgresql 12 -> 14
- Neo4j pinned to 5.2.0
- Nuxt.js 2.5 -> 3.0
- Pinia for state management (replaces Vuex)
- Vee-Validate 3 -> 4
- Tailwind 2.2 -> 3.2
- Authentication refresh token tables and schemas for long-term issuing of a new JWT access token.

[![API docs](https://raw.githubusercontent.com/whythawk/full-stack-fastapi-postgresql/master/img/landing.png)](https://github.com/whythawk/full-stack-fastapi-postgresql)

I also rebuilt the auth 'n auth to make it a bit more reliable. I haven't rebuilt the tests yet.

[Here's instructions for use](https://github.com/whythawk/full-stack-fastapi-postgresql/blob/master/README.md#how-to-use-it) and I'd appreciate your trying it out and letting me know if it is helpful.

_Projects it was used for:_
- [openLocal.uk](https://openlocal.uk) - visual data explorer for a quarterly-updated commercial location database for England and Wales.
- [Qwyre.com](https://qwyre.com) - online ereader and collaborative self-publishing.

Next project for the rebuild: [Enqwyre.com](https://eoscfuture-grants.eu/meet-the-grantees/implementation-no-code-method-schema-schema-data-transformations-interoperability) - implementation of a "no code" method for schema-to-schema data transformations for interoperability.

**Comment by br3ndonland** [comment]
Thanks @turukawa. I'd second that - if you've been stuck waiting for updates to this Cookiecutter, give the [fork](https://github.com/whythawk/full-stack-fastapi-postgresql) a try. It's an updated stack, and many of the bugs reported here have been resolved.

**Comment by podaga** [comment]
Thanks a million for this brilliant work.

Just a small typo in .../backend/app/app/core/config.py

Line 11 should read : ACCESS_TOKEN_EXPIRE_SECONDS and not ACCESS_TOKEN_EXPIRE_MINUTES

**Comment by turukawa** [comment]
Updated @podaga ... thanks. 

I've been working on extending the authentication system to include TOTP and "magic link" logins, with fall-back to passwords.

![magic-link-login](https://user-images.githubusercontent.com/998523/211525028-a60b8635-394e-4b69-a35c-4ad2a40d04b2.png)

![totp-enable-verification](https://user-images.githubusercontent.com/998523/211524852-880d7f04-e915-4bba-8730-67e1bc949327.png)

Let me know if this is of interest?

**Comment by dazzag24** [comment]
+1 for the TOTP work.

**Comment by turukawa** [comment]
[Version 0.7.0](https://github.com/whythawk/full-stack-fastapi-postgresql/releases/tag/0.7.0) released:

- New feature: magic (email-based) login, with password fallback
- New feature: Time-based One-Time Password (TOTP) authentication
- Security enhancements to improve consistency, safety and reliability of the authentication process (see full description in the frontend app)
- Refactoring of login APIs
- Requires one new frontend dependency: [QRcode.vue](https://github.com/scopewu/qrcode.vue)
